mlx/c/compile.cpp

```diff
@@ -47,6 +47,57 @@ extern "C" int mlx_detail_compile_erase(uintptr_t fun_id) {
   }
   return 0;
 }
+extern "C" int mlx_detail_compile_replace(
```

Comment:
I don’t think we need to bind all of these in the detail namespaces. Is it easy to remove them?